### ISSUE TITLE
Can't import file without an `id` column in V4

### ISSUE THREAD

**Issue by cdubz** [django-import-export/django-import-export#1812]
**Describe the bug**

In V4 I get an error attempting to import data without an `id` column.

**To Reproduce**

When updating to V4 it seems I'm no longer able to import data without an `id` field present in the import data. My expectation (and I think this was the V3 behavior?) is that if `id` is present it is used to check for an existing record but if it's not present a new record  is added.

Example configuration:

```python
class ImportExportResourceBase(resources.ModelResource):
    id = fields.Field(attribute="id")
    child = fields.Field(attribute="child_id", column_name="child_id")
    child_first_name = fields.Field(attribute="child__first_name", readonly=True)
    child_last_name = fields.Field(attribute="child__last_name", readonly=True)

    class Meta:
        clean_model_instances = True
        exclude = ("duration",)


class WeightImportExportResource(ImportExportResourceBase):
    class Meta:
        model = models.Weight


@admin.register(models.Weight)
class WeightAdmin(ImportExportMixin, ExportActionMixin, admin.ModelAdmin):
    list_display = (
        "child",
        "weight",
        "date",
    )
    list_filter = ("child", "tags")
    search_fields = (
        "child__first_name",
        "child__last_name",
        "weight",
    )
    resource_class = WeightImportExportResource
    import_error_display = ('message', 'traceback', 'row')
```

And import file:

```csv
child_id,weight,date,notes
1,10.45,2020-02-14,
1,10.17,2020-02-07,Medical that leader line general marriage plan present. Good garden suggest no. Section than order little share property why. Prepare production imagine commercial seek teach over anything. Reason late consumer.
1,10.04,2020-01-31,Drive usually end than forget. Option value either building car minute the. After resource worker response if. Include second property finish. Music event pick give wind.
1,9.89,2020-01-24,
1,9.68,2020-01-17,
```

When I try to import this file I get errors (for all rows):

```
Errors

    The following fields are declared in 'import_id_fields' but are not present in the file headers: id

    Traceback (most recent call last):
    File "/workspaces/babybuddy/.venv/lib/python3.12/site-packages/import_export/resources.py", line 871, in import_data_inner
    self._check_import_id_fields(dataset.headers)
    File "/workspaces/babybuddy/.venv/lib/python3.12/site-packages/import_export/resources.py", line 1144, in _check_import_id_fields
    raise exceptions.FieldError(
    import_export.exceptions.FieldError: The following fields are declared in 'import_id_fields' but are not present in the file headers: id
    Line number: 1 - "Column 'id' not found in dataset. Available columns are: ['child_id', 'weight', 'date', 'notes']"
    1, 10.45, 2020-02-14,

    Traceback (most recent call last):
    File "/workspaces/babybuddy/.venv/lib/python3.12/site-packages/import_export/fields.py", line 82, in clean
    value = row[self.column_name]
    ~~~^^^^^^^^^^^^^^^^^^
    KeyError: 'id'

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
    File "/workspaces/babybuddy/.venv/lib/python3.12/site-packages/import_export/resources.py", line 707, in import_row
    instance, new = self.get_or_init_instance(instance_loader, row)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/workspaces/babybuddy/.venv/lib/python3.12/site-packages/import_export/resources.py", line 177, in get_or_init_instance
    instance = self.get_instance(instance_loader, row)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/workspaces/babybuddy/.venv/lib/python3.12/site-packages/import_export/resources.py", line 170, in get_instance
    return instance_loader.get_instance(row)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/workspaces/babybuddy/.venv/lib/python3.12/site-packages/import_export/instance_loaders.py", line 29, in get_instance
    params[field.attribute] = field.clean(row)
    ^^^^^^^^^^^^^^^^
    File "/workspaces/babybuddy/.venv/lib/python3.12/site-packages/import_export/fields.py", line 84, in clean
    raise KeyError(
    KeyError: "Column 'id' not found in dataset. Available columns are: ['child_id', 'weight', 'date', 'notes']"
```

**Versions (please complete the following information):**

 - Django Import Export: 4.0.0
 - Python: 3.12
 - Django: 5.0.4

**Expected behavior**

If there is no `id` column in the data to import, new entries are added to the database for each row.

**Comment by matthewhegarty** [comment]
Thanks for raising.  Is your use case that you want to always create new records, and there is never any need to check the db for existing rows?

If you need a workaround, this may work (haven't tested) if you declare it on your resource:

```python
   def before_import(self, dataset, **kwargs):
       super().before_import(dataset, **kwargs)
       dataset.append_col([], header='id')
```

**Comment by cdubz** [comment]
> Is your use case that you want to always create new records, and there is never any need to check the db for existing rows?

My use case is that I want to create new records if there is no `id` field in the imported data or update existing records if there is an `id` field. This is how it worked in V3.

I should be able to use the `before_import` approach for the use case (by also checking for the `id` header in the dataset) but this use case feels like the most common/expected one to me. If the new behavior in V4 is necessary this change should probably be called out pretty prominently in the release/upgrade notes.

**Comment by matthewhegarty** [comment]
Thanks for raising this.  I've added a fix in #1813 to restore the v3 behaviour.

**Comment by cdubz** [comment]
Thanks for the fix and for maintaining this wonderful tool 🙏
